### PR TITLE
Json pointer replace

### PR DESCRIPTION
**Pull request by neil-stephen-mcgonigle**

### CODE REVIEW

**Comment by neil-stephen-mcgonigle** [comment]
best way to test this?

File: questionnaire/questionnaire-service.js [hunk 1]
```diff
@@ -147,7 +147,10 @@ function createQuestionnaireService({
                 const notifyService = createNotifyService({logger});
 
                 if (task.type === 'sendEmail') {
-                    if (pointer.has(questionnaire, task.templatePlaceholderMap.emailAddress)) {
+                    if (
```
Comment (sinclairs) [inline questionnaire/questionnaire-service.js:150]: Can we cache the value as per the replace-json-pointer "has" check?

File: questionnaire/questionnaire-service.js [hunk 2]
```diff
@@ -167,7 +170,10 @@ function createQuestionnaireService({
                 }
 
                 if (task.type === 'sendSms') {
-                    if (pointer.has(questionnaire, task.templatePlaceholderMap.phoneNumber)) {
+                    if (
+                        pointer.get(questionnaire, task.templatePlaceholderMap.phoneNumber) !==
```
Comment (sinclairs) [inline questionnaire/questionnaire-service.js:174]: As above, can we cache the value as per the replace-json-pointer "has" check?

**Review by sinclairs — CHANGES_REQUESTED** [review]
Looks good.

When we're happy, before merging, can you remove the "bump version to 3.0.2" commit and instead run `npm version patch`. It'll create the version commit for you and also make the corresponding change in `package-lock.json`'s version too

File: questionnaire/questionnaire-service.js (hunk 1 above)
Comment (neil-stephen-mcgonigle) [reply questionnaire/questionnaire-service.js:150]: np

File: questionnaire/questionnaire-service.js (hunk 2 above)
Comment (neil-stephen-mcgonigle) [reply questionnaire/questionnaire-service.js:174]: np

**Comment by neil-stephen-mcgonigle** [comment]
> Looks good.
> 
> When we're happy, before merging, can you remove the "bump version to 3.0.2" commit and instead run `npm version patch`. It'll create the version commit for you and also make the corresponding change in `package-lock.json`'s version too

cool

File: package-lock.json [hunk 3]
```diff
@@ -2486,6 +2486,14 @@
                 "json-to-ast": "^2.0.3",
                 "jsonpointer": "^4.0.1",
                 "leven": "^3.1.0"
+            },
```
Comment (sinclairs) [inline package-lock.json:2489]: File is missing version bump

**Comment by sinclairs** [comment]
> 
> 
> best way to test this?

I'd hope the current tests cover this, you reckon we need something else?

**Comment by neil-stephen-mcgonigle** [comment]
> > best way to test this?
> 
> I'd hope the current tests cover this, you reckon we need something else?
Happy with that just wanted assurance.

**Review by sinclairs — CHANGES_REQUESTED** [review]
Nice one. Just need to sort the commit order and we're good.

Current order: 
- Replace json-pointer module with jsonpointer
- 3.0.3
- caching calls to pointer.get 

Can we squash? 
- Replace json-pointer module with jsonpointer
- caching calls to pointer.get 

Proposed order (oldest to newest):
- Replace json-pointer module with jsonpointer
- 3.0.3 

We can then tag the 3.0.3 as the release

**Review by sinclairs — APPROVED** [review]

**Pull request merged**